### PR TITLE
Add balance command

### PR DESCRIPTION
**Pull request by jrchatruc**
# What :computer: 
* Add `npx zksync-cli wallet balance` command to check ETH balance of the specified address

# Why :hand:
* Seeing ETH balance might be useful during the development

# Evidence :camera:
image

### CODE REVIEW

File: src/commands/wallet/command.ts [hunk 1]
```diff
@@ -0,0 +1,3 @@
+import Program from "../../program.js";
+
+export default Program.command("wallet").description("Manage wallet related features for L2 and L1");
```
Comment (JackHamer09) [inline src/commands/wallet/command.ts:3]: For now there is no L1 features.
So we should either not mention L1 and just say `Manage zkSync wallet related features`
or
Add possibility to choose L1 network in the list. For example:
```
❯ zkSync Era Testnet 
  zkSync Era Mainnet 
  Local Dockerized node
  ---
  Ethereum Mainnet
  Ethereum Goerli Testnet
```

File: src/commands/wallet/balance.ts [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+        options = {
+            ...options,
+            ...answers,
+        };
+
+        const toChain = l2Chains.find((e) => e.network === options.chain);
+        const Provider = getL2Provider(options.l2RpcUrl ?? toChain!.rpcUrl);
+        const balance = await Provider.getBalance(options.account ?? "Unknown account");
+
+        Logger.info(`\n${bigNumberToDecimal(balance)} ETH`);
```
Comment (JackHamer09) [inline src/commands/wallet/balance.ts:58]: Let's say `${selectedChain.name} Balance: XX ETH`

File: src/commands/wallet/balance.ts [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+            ],
+            options
+        );
+
+        options = {
+            ...options,
+            ...answers,
+        };
+
+        const toChain = l2Chains.find((e) => e.network === options.chain);
```
Comment (JackHamer09) [inline src/commands/wallet/balance.ts:54]: Let's call it `selectedChain`

File: src/commands/wallet/balance.ts [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+        const Provider = getL2Provider(options.l2RpcUrl ?? toChain!.rpcUrl);
```
Comment (JackHamer09) [inline src/commands/wallet/balance.ts:55]: Let's not create uppercase variables

File: src/commands/wallet/balance.ts [hunk 5]
```diff
@@ -0,0 +1,74 @@
+import inquirer from "inquirer";
+
+import Program from "./command.js";
+import {
+    accountOption,
```
Comment (JackHamer09) [inline src/commands/wallet/balance.ts:5]: Try to properly setup prettier so the code is properly styled

File: src/common/options.ts [hunk 6]
```diff
@@ -26,6 +27,12 @@ export type DefaultTransactionOptions = DefaultOptions & {
   l2RpcUrl?: string;
   privateKey: string;
 };
+export type BalanceOptions = DefaultOptions & {
```
Comment (JackHamer09) [inline src/common/options.ts:30]: I don't think we should keep `BalanceOptions` in this file. Let's move it to balance.ts where it is used.

**Review by JackHamer09 — CHANGES_REQUESTED** [review]
Should we also ask for which token to get balance to? ETH or ERC20
And then if ERC20 is selected we also ask for address of the token.
Would be useful to have something like that :)
Not required for this PR tho..

**Comment by jrchatruc** [comment]
> Should we also ask for which token to get balance to? ETH or ERC20 And then if ERC20 is selected we also ask for address of the token. Would be useful to have something like that :) Not required for this PR tho..

That would be very useful indeed, we'll add it on a subsquent PR!

**Comment by jrchatruc** [comment]
Addressed all the commends above here [aef66ef](https://github.com/matter-labs/zksync-cli/pull/79/commits/aef66ef45749923c48763bc356c0c36a87e32b41) and here [2e88921](https://github.com/matter-labs/zksync-cli/pull/79/commits/2e8892186e7e3538e4433467d4dae6970b90cf63). In regards to this

```
For now there is no L1 features.
So we should either not mention L1 and just say Manage zkSync wallet related features
or
Add possibility to choose L1 network in the list.
```

For now I just changed the description to not mention L1, we might add L1 features in the future.

**Pull request closed without merging**